in-vehicle-digital-twin/src/main.rs

```diff
@@ -26,13 +24,11 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     // Setup the HTTP server.
     let addr: SocketAddr = IN_VEHICLE_DIGITAL_TWIN_ADDR.parse()?;
-    let provider_impl = provider_impl::ProviderImpl::default();
-    let digitaltwin_impl =
-        digitaltwin_impl::DigitalTwinImpl { entity_map: Arc::new(RwLock::new(HashMap::new())) };
-    let server_future = Server::builder()
-        .add_service(ProviderServer::new(provider_impl))
-        .add_service(DigitalTwinServer::new(digitaltwin_impl))
-        .serve(addr);
+    let digitaltwin_impl = digitaltwin_impl::DigitalTwinImpl {
+        entity_access_info_map: Arc::new(RwLock::new(HashMap::new())),
```

Comment:
we aren't sharing this data so I don't think it needs to be wrapped in an `Arc`. We are only using the mutex functionality and not the reference counting for shared state. Can we refactor this property to just be `RwLock<HashMap<...>>`?

Reply:
We want a reference count, not the object itself.